### PR TITLE
Move TermAndBoost back to its original location.

### PR DESCRIPTION
**Pull request by jpountz**
PR #12169 accidentally moved the `TermAndBoost` class to a different location, which would break custom sub-classes of `QueryBuilder`. This commit moves it back to its original location and gives word2vec its own TermAndBoost class.

### CODE REVIEW

**Comment by reta** [comment]
Thanks @jpountz , it actually did break (OpenSearch) and I suspect Elasticsearch as well

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java [hunk 1]
```diff
@@ -14,7 +14,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.lucene.util;
+package org.apache.lucene.analysis.synonym.word2vec;
+
+import org.apache.lucene.util.BytesRef;
```
Comment (iverase) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: Should we make this class package private?

Comment (iverase) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: And even rename it so it does not get confused?

Comment (jpountz) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: Good call on reducing visibility. I tried to see if private worked and it did not, but I forgot to check if pkg-private would work, it does!

I think it's fine to use the same name, at least I don't want to give it a weird name for the sake of having a different name? E.g. we have lots of `ScoreAndDoc` classes across many packages and it's fine?

Comment (iverase) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: it makes sense, thanks!

Comment (iverase) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: I just noticed that `Word2VecSynonymProvider` is a public class the exposes `TermAndBoost` via the method `#getSynonyms`. So either we can make that method package private or we need to change the visibility of the class again.

**Review by uschindler — APPROVED** [review]

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java (hunk 1 above)
Comment (uschindler) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/word2vec/TermAndBoost.java:19]: I would keep it public for now as it was before.

**Review by iverase — APPROVED** [review]

**Pull request merged**